tax_functions.js

```diff
@@ -1,27 +1,48 @@
+
+const incomeTaxBrackets = {
+  'S': [
+    {threshold: 0,      rate: 0.10},
+    {threshold: 9951,   rate: 0.12},
+    {threshold: 40526,  rate: 0.22},
+    {threshold: 86376,  rate: 0.24},
+    {threshold: 164926, rate: 0.32},
+    {threshold: 209426, rate: 0.35},
+    {threshold: 523601, rate: 0.37},
+  ],
+  'MFJ': [
+    {threshold: 0,      rate: 0.10},
+    {threshold: 19901,  rate: 0.12},
+    {threshold: 81051,  rate: 0.22},
+    {threshold: 172751, rate: 0.24},
+    {threshold: 329851, rate: 0.32},
+    {threshold: 418851, rate: 0.35},
+    {threshold: 628301, rate: 0.37},
+  ],
+};
+
 // Given some income, return the income tax amount. Based on 2021 rates.
 function INCOMETAX(income, filingStatus) {
-  if (filingStatus !== 'MFJ') {
-    throw new Error('filingStatus currently only supports MFJ');
-  }
-  if (income <= 19900) {
-    return income * 0.10;
-  }
-  if (income <= 81050) {
-    return 1990 + (income - 19900) * 0.12;
-  }
-  if (income <= 172750) {
-    return 9328 + (income - 81050) * 0.22;
+  const brackets = incomeTaxBrackets[filingStatus];
+  if (!brackets) {
+    throw new Error(`"${filingStatus}" not supported as a filing status`);
```

Comment:
This works well but it might also be nice to mention the valid filing statuses. (`Object.keys(incomeTaxBrackets)`).